lib/request/TingClientSuggestionRequest.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+class TingClientSuggestionRequest extends TingClientRequest {
+
+  protected $query;
+  protected $index;
+  protected $facetIndex;
+  protected $filterQuery;
+  protected $sort;
+  protected $agency;
+  protected $profile;
+  protected $maxSuggestions;
+  protected $maxTime;
+  protected $highlight;
+  protected $highlightPre;
+  protected $highlightPost;
+  protected $logger;
+
+  public function getRequest() {
+    $this->setParameter('action', 'sug:getFacet');
+
+    $methodParameterMap = array(
+      'index' => 'index',
+      'query' => 'query',
+      'facetIndex' => 'facetIndex',
+      'filterQuery' => 'filterQuery',
+      'agency' => 'agency',
+      'profile' => 'profile',
+      'highlight' => 'highlight',
+      'highlightPre' => 'highlight.pre',
+      'highlightPost' => 'highlight.post',
+      'maxSuggestions' => 'maxSuggestions',
+      'maxTime' => 'maxTime',
+      'sort' => 'sort',
+      'outputType' => 'json',
+    );
+
+    foreach ($methodParameterMap as $method => $parameter) {
+      $getter = 'get' . ucfirst($method);
+      if (method_exists('TingClientSuggestionRequest', $getter)) {
+        if ($value = $this->$getter()) {
+          $this->setParameter($parameter, $value);
+        }
+      }
+      else {
+        $this->logger = new TingClientDrupalWatchDogLogger();
+        $this->logger->doLog('Call to undefined method TingClientSuggestionRequest::' . $getter, WARNING);
```

Comment:
I understand the intent here, but I would prefer if we could avoid hardcoding Drupal-specific classes into the library code.

1. It makes it even harder to make it framework agnostic. If we want to use a logger then we should inject a logger class and use that.
2. As I see it the current approach to logging provides no value. It checks whether we have typed the right names into the map. This should be pretty trivial to check. If we actually have not we risk missing it because the error may be hiding in the watchdog log.